django_rq/utils.py

```diff
@@ -10,9 +10,42 @@
 from rq.worker import Worker
 from rq.worker_registration import clean_worker_registry
 
-from .queues import get_connection, get_queue_by_index
+
+from .queues import get_connection, get_queue_by_index, get_scheduler
 from .settings import QUEUES_LIST
 from .templatetags.django_rq import to_localtime
+from django.core.exceptions import ImproperlyConfigured
+
+def get_scheduler_pid(queue):
+    '''Checks whether there's a scheduler-lock on a particular queue, and returns the PID.
+        It works by first checking if there's an RQ-Scheduler active.
+        If not, it checks the RQ's RQScheduler for a scheduler lock in the desired queue
+        Note: result might have some delay (1-15 minutes) but it helps visualizing whether the setup is working correcly
+    '''
+    try:
+        if not queue:
+            raise ValueError("queue argument not defined")
+
+        # first try to use rq-scheduler
+        scheduler = get_scheduler(queue.name)  # should fail if rq_scheduler not present
+        # currently, scheduler_lock_key is not used but, just in case, try
+        lock_key = scheduler.scheduler_lock_key
+        conn = scheduler.connection
+        if _ := conn.get(lock_key):
+            return scheduler.pid
+        else:
+            for key in conn.keys(f"{scheduler.redis_scheduler_namespace_prefix}*"):
```

Comment:
I just realized that running `keys()` will be very costly in a Redis database with a large number of keys. I think we should avoid doing this and just support the built in `RQScheduler`.

Reply:
Yes, I agree totally. Perhaps making a PR to make sure rq-scheduler sets scheduler_lock_key? because there will be a transition period. for now, I will make it so if no scheduler_lock_key, will return -1 and will be rendered as "Unknown". Will let you know after I finish fixing the PR.

Reply:
Finally, I decided to just omit functionality if RQ-Scheduler is being used as per your recommendation.

Reply:
See below, functionality was removed for rq-scheduler.